### ISSUE TITLE
Kuma reports unhealthy docker containers as "Pending", not "Down"

### ISSUE THREAD

**Issue by vfosnar** [louislam/uptime-kuma#4369]
### ⚠️ Please verify that this question has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

As the title says, is there any way to make Kuma treat unhealthy containers as downtime?

![image](https://github.com/louislam/uptime-kuma/assets/13472046/3a4b4267-9d9b-49c7-8ed8-680852adcafa)

### 🐻 Uptime-Kuma Version

1.23.11

### 💻 Operating System and Arch

Docker Compose

### 🌐 Browser

unrelated

### 🐋 Docker Version

Docker 24.0.7, build afdd53b

**Comment by chakflying** [comment]
This might be a bug. I assume you have not configured a large number of retries?

**Comment by vfosnar** [comment]
Yes.

I also tried it with the default retries = 0, resend notification if down = 0

![image](https://github.com/louislam/uptime-kuma/assets/13472046/6871af24-0611-4196-91b3-cf87cb02b0b3)

**Comment by vfosnar** [comment]
Reproduced the error on a clean Kuma install. If I hover over the yellow line it says "unhealthy"

![image](https://github.com/louislam/uptime-kuma/assets/13472046/3954fefb-e43a-458c-a63f-66934c83caed)

**Comment by rmatte** [comment]
I saw this today where one of my containers was unhealthy but it never generated an event or notification for it. I have retries set at zero, so this should have alerted.

![Screenshot 2024-12-02 124407](https://github.com/user-attachments/assets/6c07c390-fab4-47b4-8861-bd72cc0f6751)

**Comment by charliik** [comment]
up

![image](https://github.com/user-attachments/assets/04709296-7029-4212-9348-94298a9a2486)

Got no notifications whatsoever while my service was unhealthy

**Comment by Iceman248** [comment]
I just came across this, had container that was not working and it was stuck at pending, didn't alert me.

**Comment by mikehansen20** [comment]
Running into the same issue here.  No alert triggers for a container having issues. It just hangs on "Pending" indefinitely.

**Comment by andrew-manger** [comment]
Same here. I'd like to be able to configure notifications for unhealthy containers. I think it should be pending (preferably "starting") as the healthcheck waits out its startup period and then should report healthy, unhealthy, or down.

**Comment by vivami** [comment]
Same issue here, status goes to "PENDING" when container gets into the "unhealthy" state.

**Comment by sethsteenken** [comment]
Same issue. Container is monitored and records as unhealthy. Container monitoring should have the option to specify "unhealthy" status to be considered DOWN and a notification is sent.

**Comment by lzecca78** [comment]
Same issue here, kuma keeps consindering the container `pending` even if the threshold has been passed:

```
2025-04-13T20:44:25+02:00 [MONITOR] WARN: Monitor #45 'dozzle': Pending: Container State is exited | Max retries: 2 | Retry: 1 | Retry Interval: 60 seconds | Type: docker
```
I've stopped the container at 20:35 with 2 retries, each every 60 seconds. But the `Retry` count stuck at 1 `Retry: 1` and the counter is not increased.

**Comment by mcanning** [comment]
Just saw this today as well, definitely unexpected behavior

**Comment by sahara101** [comment]
Same issue

**Comment by Pytonballoon810** [comment]
Any news on the issue? This is extra annoying when using docker containers to monitor other stuff and leveraging the container healthcheck as an indicator. Any way for a workaround?

**Comment by CommanderStorm** [comment]
The associated pr is blocked by not knowing how to treat starting.
If you have on-topic, non-duplicate, no +1 comments, please raise them there.

**Comment by rmatte** [comment]
For "starting" state I think the solution is to make the minimum number of retries for this check type be 1 and have it alert if it's in starting state twice in a row which would generally be unusual for a docker container. People can tweak the timing and retries beyond that as needed. I can't really think of any better way to address this. It will be significantly better than how it's currently behaving.

**Comment by CommanderStorm** [comment]
There has been another PR.
I don't currently have the time to test it, so would appreciate someone from the community to give it a testrun:

- https://github.com/louislam/uptime-kuma/pull/6292

Here are our docs how a PR can be taken for a spin: https://github.com/louislam/uptime-kuma/wiki/Test-Pull-Requests

**Comment by mathis6787** [comment]
Same issue here, any update ?

**Comment by CommanderStorm** [comment]
@mathis6787  could you test the PR I mentioned above? I still am very buisy on other stuff, but if that is truely better (not just a sidegrade), I would be happy to approve it.


Needs to be actually better though. I don't want to churn on this every second month.

**Comment by mathis6787** [comment]
Hi @CommanderStorm Sorry I didn't Read your last comment last,  my bad.  I wish I could do it but I never worked with that kind this kind of  tech stack I dont feel legitimate to review.  If I learn the stack in the future I will be happy to contribute asap.

**Comment by collyncamara** [comment]
> [@mathis6787](https://github.com/mathis6787) could you test the PR I mentioned above? I still am very buisy on other stuff, but if that is truely better (not just a sidegrade), I would be happy to approve it.
> 
> Needs to be actually better though. I don't want to churn on this every second month.

I can't comment on anything other than this added the desired functionality for me. Like @mathis6787 I'm very new to testing and commenting on PRs and issues, but I was able to get the pr-test2 image up and running with the PR you linked. Sent my container into an unhealthy state and Kuma quickly sent a notification that the service was down.

Anything else need to be tested/validated? Appreciate the eyes on this!!
